reader.go

```diff
@@ -1422,6 +1427,29 @@ func (r *Reader) ReadLag(ctx context.Context) (lag int64, err error) {
 	return
 }
 
+func (r *Reader) ConsumerGroupOffsets() (map[int]int64, error) {
+	if !r.useConsumerGroup() {
+		return nil, fmt.Errorf("only supported for consumer groups")
+	}
+
+	// establish a connection to the coordinator.  this connection will be
```

Comment:
This is nitpicky, but I'd prefer to see the connection to the coordinator below where you actually use it.

Reply:
@stevevls can do, just to be clear though there is a small trade-off there, that is if the `r.coordinator()` errors we would needlessly lock the reader to make the copy (which won't get used). Total minor nit as well, but just to be clear. If you're still +1 I'll move it down.

Reply:
Yeah, I dig that.  My hope would be that no one is looping on this so quickly as to cause contention on the lock. 🤞 If they are, then that should be the least of their worries.  I'll leave it up to you since my suggestion was for readability reasons.